src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -762,6 +796,81 @@ internal static void ValidateSpecifiedAuthentication(PSCredential credential, st
+        private const string HostNameAlias = "HostName";
+        private const string UserNameParameter = "UserName";
+        private const string KeyFilePathParameter = "KeyFilePath";
+        private const string IdentityFilePathAlias = "IdentityFilePath";
+        internal SSHConnection[] ParseSSHConnectionHashTable()
+        {
+            List<SSHConnection> connections = new List<SSHConnection>();
+            foreach (var item in this.SSHConnection)
+            {
+                if (item.ContainsKey(ComputerNameParameter) && item.ContainsKey(HostNameAlias))
+                {
+                    throw new PSArgumentException(RemotingErrorIdStrings.SSHConnectionDuplicateHostName);
+                }
+
+                if (item.ContainsKey(KeyFilePathParameter) && item.ContainsKey(IdentityFilePathAlias))
+                {
+                    throw new PSArgumentException(RemotingErrorIdStrings.SSHConnectionDuplicateKeyPath);
+                }
+
+                SSHConnection connectionInfo = new SSHConnection();
+                foreach (var key in item.Keys)
+                {
+                    string paramName = key as string;
+                    if (string.IsNullOrEmpty(paramName))
+                    {
+                        throw new PSArgumentException(RemotingErrorIdStrings.InvalidSSHConnectionParameter);
+                    }
+
+                    string paramValue = item[paramName] as string;
+                    if (string.IsNullOrEmpty(paramValue))
+                    {
+                        throw new PSArgumentException(RemotingErrorIdStrings.InvalidSSHConnectionParameter);
+                    }
+
+                    if (paramName.Equals(ComputerNameParameter, StringComparison.OrdinalIgnoreCase) || paramName.Equals(HostNameAlias, StringComparison.OrdinalIgnoreCase))
+                    {
+                        var resolvedComputerName = ResolveComputerName(paramValue);
+                        ValidateComputerName(new string[] { resolvedComputerName });
+                        connectionInfo.ComputerName = resolvedComputerName;
+                    }
+                    else if (paramName.Equals(UserNameParameter, StringComparison.OrdinalIgnoreCase))
+                    {
+                        connectionInfo.UserName = paramValue;
+                    }
+                    else if (paramName.Equals(KeyFilePathParameter, StringComparison.OrdinalIgnoreCase) || paramName.Equals(IdentityFilePathAlias, StringComparison.OrdinalIgnoreCase))
+                    {
+                        connectionInfo.KeyFilePath = paramValue;
+                    }
+                    else
+                    {
+                        throw new PSArgumentException(
+                            StringUtil.Format(RemotingErrorIdStrings.UnknownSSHConnectionParameter, paramName));
+                    }
+                }
+
+                if (string.IsNullOrEmpty(connectionInfo.ComputerName))
```

Comment:
It is possible for ComputerName or HostName to exist but contain only space characters.  So the check would have to be a contains coupled with a not-empty check.  It is easier to do the single check here.